.github/workflows/jacoco.yml

```diff
@@ -0,0 +1,51 @@
+name: Coverage
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Set up JDK 11 for Autogen
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 11
+          cache: 'maven'
+      - uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: shardingsphere-maven-third-party-e2e-cache-${{ github.sha }}
+          restore-keys: |
+            shardingsphere-maven-third-party-e2e-cache-
+            shardingsphere-maven-third-party-
+
+      - name: Build Maven Env for Autogen
+        run: ./mvnw clean install -T1C
+
+      - name: Set up JDK 8 for Jacoco
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 8
+          cache: 'maven'
+
+      - name: Build with Maven on Jacoco-exec
+        run: ./mvnw --show-version --batch-mode --no-transfer-progress verify -Dmaven.test.failure.ignore=true -DfailIfNoTests=false -Dsurefire.failIfNoSpecifiedTests=false
+
+      - name: Make script executable
+        run: chmod +x ./.github/workflows/resources/scripts/unit-test-coverage-merge/code-coverage-merge.sh
+
+      - name: Run code coverage merge script
+        run: ./.github/workflows/resources/scripts/unit-test-coverage-merge/code-coverage-merge.sh
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          files: ./target/mergeReport/jacoco.xml
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
codecov not supported tokenless anymore
this should contact Apache infra team.

Reply:
I agree with that. Because I ever tried to use token less for codecov but it give me the error which can be checked on this [link](https://github.com/AkaLrz/shardingsphere/actions/runs/9948671873/job/27483668717). I thought we should create a tikect on Apache infra team for this Codecov action.

Reply:
And also I think if we add `if: github.repository == 'apache/shardingsphere'` under the with, it will help us to make github action catch token quickly?

Reply:
https://issues.apache.org/jira/browse/INFRA-26019?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel

Done.

Reply:
> And also I think if we add `if: github.repository == 'apache/shardingsphere'` under the with, it will help us to make github action catch token quickly?

this will not work. token is a special number on codecov settings. It should be set as a github action variable

Reply:
Got it. I removed `if: github.repository == 'apache/shardingsphere'`